### ISSUE TITLE
Running archetype to generate maven lambda is failing

### ISSUE THREAD

**Issue by michaelgman13** [aws/aws-sdk-java-v2#1981]
## Describe the bug

I'm running the following command to try to generate a lambda maven project. `mvn "archetype:generate" "-DarchetypeGroupId=software.amazon.awssdk" "-DarchetypeArtifactId=archetype-lambda" "-DarchetypeVersion=2.11.4"  "-DgroupId=com.amazon.test" "-DartifactId=test-project" "-Dservice=s3" "-Dregion=us-west-2" "-DinteractiveMode=false" -e` 

## Expected Behavior

I think it should create the maven project successfully. Was following [this page](https://aws.amazon.com/blogs/developer/bootstrapping-a-java-lambda-application-with-minimal-aws-java-sdk-startup-time-using-maven/)

## Current Behavior

I got this error output:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.2.0:generate (default-cli) on project standalone-pom: Error merging velocity templates: Unable to find resource 'archetype-resources/global.vm' -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.2.0:generate (default-cli) on project standalone-pom: Error merging velocity templates: Unable to find resource 'archetype-resources/global.vm'
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoFailureException: Error merging velocity templates: Unable to find resource 'archetype-resources/global.vm'
    at org.apache.maven.archetype.mojos.CreateProjectFromArchetypeMojo.execute (CreateProjectFromArchetypeMojo.java:208)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

## Steps to Reproduce

Run mvn `"archetype:generate" "-DarchetypeGroupId=software.amazon.awssdk" "-DarchetypeArtifactId=archetype-lambda" "-DarchetypeVersion=2.11.4"  "-DgroupId=com.michael.test" "-DartifactId=test-project" "-Dservice=s3" "-Dregion=us-west-2" "-DinteractiveMode=false" -e` in an empty directory

## Context

I'd like to use this archetype to generate lambda templates. 

## Your Environment

* AWS Java SDK version used: N/A since I'm using a maven command, but aws --version returns aws-cli/2.0.38 Python/3.7.7 Windows/10 exe/AMD64
* JDK version used: 12
* Operating System and version: windows 10

```
mvn --version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\Program Files\apache-maven-3.6.3\bin\..
Java version: 12, vendor: Oracle Corporation, runtime: C:\Program Files\jdk-12
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

**Comment by zoewangg** [comment]
Hi @michaelgman13 , I was not able to reproduce the issue. I found this answer on stackoverflow and thought it could be related https://stackoverflow.com/a/26758522

Could you check if you have write permission of the directory?

**Comment by debora-ito** [comment]
@michaelgman13 were you able to try @zoewangg 's suggestion?

**Comment by nchauhan5** [comment]
@debora-ito , @michaelgman13  - I am facing the same issue. The exact same error logs mentioned above with -e flag. I have permission to read and write in the folder, I am running command prompt in admin mode so permissions should not be a problem here.

**Comment by zoewangg** [comment]
Hi @nchauhan5, did you also run the command on windows?

**Comment by nchauhan5** [comment]
@zoewangg  Yes

**Comment by zoewangg** [comment]
@jason-riddle Thank you for providing detailed steps to reproduce the issue. I'm able to reproduce it using the docker image. We will look into the issue

**Comment by zoewangg** [comment]
I think I found out the root cause. It seems the new release of `maven-archetype-plugin` 3.2.0 broke it. I was running older version of `maven-archetype-plugin` on my local machine and that's why I couldn't reproduce it there.

In 3.2.0, the default `ResourceLoader` is `ClasspathResourceLoader`
```
[DEBUG] Default Properties File: org/apache/velocity/runtime/defaults/velocity.properties
[DEBUG] Default ResourceManager initializing. (class org.apache.velocity.runtime.resource.ResourceManagerImpl)
[DEBUG] ResourceLoader instantiated: org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader
```

In 3.0.1, it is `ContextClassLoaderResourceLoader`
```
[DEBUG] Default ResourceManager initializing. (class org.apache.velocity.runtime.resource.ResourceManagerImpl)
[DEBUG] ResourceLoader instantiated: org.codehaus.plexus.velocity.ContextClassLoaderResourceLoader
[DEBUG] ResourceLoader instantiated: org.codehaus.plexus.velocity.SiteResourceLoader
```

We will need to investigate how to fix it.

As a workaround, you can use the following command to enforce the use of older version of the archetype plugin

```
mvn org.apache.maven.plugins:maven-archetype-plugin:3.1.0:generate \
  -DarchetypeGroupId=software.amazon.awssdk \
  -DarchetypeArtifactId=archetype-lambda \
  -DarchetypeVersion=2.14.3 \
  -DgroupId=com.test \
  -DartifactId=sample-project \
  -Dservice=s3 \
  -Dregion=US_WEST_2 \
  -DinteractiveMode=false
```

**Comment by nchauhan5** [comment]
@zoewangg, what will be the command be, if instead of "archetype-lambda" archetypeArtifactId, I want to create a project with AWS Rekognition service as parts of the pom.xml

**Comment by zoewangg** [comment]
@nchauhan5 You mean an archetype for a general application rather than Lambda function? We don't have one yet. Could you create a feature request? We will prioritize it if more people +1 on it.

If you mean Lambda function, you can use the same command and change the `service` part to `-Dservice=rekognition`. You can find all available services here: https://github.com/aws/aws-sdk-java-v2/tree/master/services

**Comment by fjleon1980** [comment]
happening here as well, using windows 10. note that this example is https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/setup-project-maven.html#create-maven-project

**Comment by luankevinferreira** [comment]
Hi @zoewangg, I was with the same problem and the workaround worked!

**Comment by zoewangg** [comment]
Fixed via #2085 and released as part of `2.15.8`

With the latest version, now you don't need to specify the `maven-archetype-plugin` version any more. 

```
mvn archetype:generate \
    -DarchetypeGroupId=software.amazon.awssdk \
    -DarchetypeArtifactId=archetype-lambda \
    -DarchetypeVersion=2.15.8 \
    -DgroupId=com.amazon.test \
    -DartifactId=test-project \
    -Dservice=s3  \
    -Dregion=us-west-2 \
    -DinteractiveMode=false
```

**Comment by nkdevirl** [comment]
This blog post should be updated to stop other folks trying to find the solution for this bug

https://aws.amazon.com/blogs/developer/bootstrapping-a-java-lambda-application-with-minimal-aws-java-sdk-startup-time-using-maven/
